### PR TITLE
Mouse Buttons

### LINKED ISSUE #131: Mouse Button Events
Raise an event of some kind when a mouse button has been pressed but the cursor has not moved.

### PR DESCRIPTION
**Pull request by astronouth7303**
Implements mouse buttons.

I would like feedback as to how you feel about this style before I adjust `MouseMotion` to match.

Fixes #131.

### CODE REVIEW

File: ppb/events.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 from ppb.abc import Scene
 from ppb.vector import Vector
+from .buttons import MouseButton
```
Comment (pathunstrom) [inline ppb/events.py]: Can we use absolute imports, please?

File: ppb/systems.py [hunk 2]
```diff
@@ -55,9 +63,11 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
     def on_update(self, update, signal):
         for pygame_event in pygame.event.get():
-            ppb_event = self.event_map.get(pygame_event.type)
-            if ppb_event is not None:
-                signal(getattr(self, ppb_event)(pygame_event, update.scene))
+            methname = self.event_map.get(pygame_event.type)
+            if methname is not None:
+                ppbevent = getattr(self, methname)(pygame_event, update.scene)
+                if ppbevent:
```
Comment (pathunstrom) [inline ppb/systems.py]: This should never happen. If it's in the method map, it's on the class, and if it's not, that's on us.

File: ppb/systems.py [hunk 3]
```diff
@@ -70,48 +80,27 @@ def mouse_motion(self, event, scene):
...
+    def button_pressed(self, event, scene):
+        screen_position = Vector(*event.pos)
+        camera = scene.main_camera
+        game_position = camera.translate_to_frame(screen_position)
+        btn = self.button_map.get(event.button)
+        if btn is not None:
+            return events.ButtonPressed(
+                button=btn,
+                position=game_position,
+            )
```
Comment (pathunstrom) [inline ppb/systems.py]: We definitely need screen space on these.

File: ppb/systems.py (hunk 2 above)
Comment (astronouth7303) [reply ppb/systems.py]: That's not what this tests. This tests if the pygame event actually produced a ppb event.

File: ppb/systems.py (hunk 3 above)
Comment (astronouth7303) [reply ppb/systems.py]: 1. Why?
2. This isn't available on pyglet

**Pull request merged**